src/data/finset/basic.lean

```diff
@@ -2543,6 +2543,10 @@ lemma disjoint_filter_filter {s t : finset α} {p q : α → Prop} [decidable_pr
   (disjoint s t) → disjoint (s.filter p) (t.filter q) :=
 disjoint.mono (filter_subset _ _) (filter_subset _ _)
 
+lemma disjoint_filter_filter_neg (s : finset α) (p : α → Prop) [decidable_pred p] :
+  disjoint (s.filter p) (s.filter $ λ a, ¬ p a) :=
+disjoint_filter.2 $ λ a _, not_not.2
```

Comment:
Does this also work as
```suggestion
(disjoint_filter.2 $ λ a _, id).symm?
```

Reply:
Yup. Do you consider that an improvement?

Reply:
Does it avoid `#print axioms` giving `choice`?